setup.py

```diff
@@ -17,6 +18,15 @@
     ]
 }
 
+if sys.version_info[:2] <= (3, 11):
+    EXTRA_REQUIRES.get('dev').append(
+        "flake8~=4.0.1"
+    )
+else:
+    EXTRA_REQUIRES.get('dev').append(
```

Comment:
What is this for?  Can't we directly update flake8?

Reply:
Discussed outside of GH -- the flake8 version we have pinned to works for python 37-311 but fails for 312. There currently isn't a version that works for 37 and 312, so we have to install a specific version depending on python version. We can revisit this in the future when removing 37 tests.